### PR TITLE
Feature/likes fetch

### PR DESCRIPTION
**Pull request by fominmaksim**

### CODE REVIEW

File: src/helpers/validation.ts [hunk 1]
```diff
@@ -25,3 +28,24 @@ export const followersSchema = yup.object({
...
+    return userIdSchema;
+  } else if (values.tweetId) {
+    return tweetIdSchema;
+  } else {
+    return yup.object({
+      userId: yup.string().required("tweetId or userId are required"),
+      tweetId: yup.string().required("tweetId or userId are required"),
+    });
+  }
+});
```
Comment (alexandrtovmach) [inline src/helpers/validation.ts:38]: You don't need to make it conditional validation, just put all query params into single schema, and don't make them required

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/middlewares/yup.middlewares.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { AnyYupSchema } from "../helpers/validation";
 export const validation =
   (schema: AnyYupSchema) =>
   async (req: Request, res: Response, next: NextFunction) => {
-    const resource = req.body;
+    const resource = req.body && req.query;
```
Comment (alexandrtovmach) [inline src/middlewares/yup.middlewares.ts]: it's really strange 0_o. What happen if you'll provide body and query in the same time?

please validate full `req` object with one single schema, with next keys:
```
{
  body: ...
  query: ...
}
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/repositories/likes.repository.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+  options: object
+) => {
+  return LikeModel.paginate(query, {
+    ...options,
+    populate: ["user", "tweet"],
+  });
+};
+
 export const getOne = async (
   data: { _id: string } | { user: string; tweet: string }
```
Comment (VitaliiDenysiuk) [inline src/repositories/likes.repository.ts:11]: make normal interface

File: src/services/likes.service.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
   deleteOne,
   updateOne,
+  listLikesByQueryParams,
 } from "../repositories/likes.repository";
 import { changeTweetStats } from "./tweets.service";
 
+export const getLikes = async (data: PaginationParameters<never, never>) => {
+  const likesList = await listLikesByQueryParams(...data.get());
+  return likesList;
+};
```
Comment (VitaliiDenysiuk) [inline src/services/likes.service.ts]:
```suggestion
export const getLikes = (data: PaginationParameters<never, never>) => {
  return listLikesByQueryParams(...data.get());
};
```

File: src/helpers/validation.ts [hunk 5]
```diff
@@ -25,3 +25,8 @@ export const followersSchema = yup.object({
   follower: yup.string().required("followerId can't be blank"),
   following: yup.string().required("ownerId can't be blank"),
 });
+
+export const likesQuerySchema = yup.object({
+  tweet: yup.string(),
```
Comment (VitaliiDenysiuk) [inline src/helpers/validation.ts]: strange Schema. do you really sure that this schema needed ?

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/repositories/likes.repository.ts (hunk 3 above)
Comment (fominmaksim) [reply src/repositories/likes.repository.ts:11]: what is normal interface?

File: src/helpers/validation.ts (hunk 5 above)
Comment (alexandrtovmach) [reply src/helpers/validation.ts]: yes, but it should be updated

File: src/middlewares/yup.middlewares.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { AnyYupSchema } from "../helpers/validation";
 export const validation =
   (schema: AnyYupSchema) =>
   async (req: Request, res: Response, next: NextFunction) => {
-    const resource = req.body;
+    const resource = req;
```
Comment (alexandrtovmach) [inline src/middlewares/yup.middlewares.ts]: and do we need this variable?

**Review by alexandrtovmach — APPROVED** [review]

File: src/repositories/likes.repository.ts [hunk 7]
```diff
@@ -1,5 +1,15 @@
 import { LikeModel } from "../models/Like.model";
 
+export const listLikesByQueryParams = async (
```
Comment (alexandrtovmach) [inline src/repositories/likes.repository.ts]:
```suggestion
export const getAll = async (
```

File: src/repositories/likes.repository.ts (hunk 3 above)
Comment (alexandrtovmach) [reply src/repositories/likes.repository.ts:11]: I think @VitaliiDenysyuk mean to not define types in place, and have separate type/interface defined

**Review by alexandrtovmach — APPROVED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**